lib/rspec/support/caller_filter.rb

```diff
@@ -73,10 +73,24 @@ def self.first_non_rspec_line(skip_frames=3, increment=5)
         end
       end
     else
+      CallerFinder = Class.new(StandardError)
+      class LazyBacktrace
+        def initialize(exception)
+          @exception = exception
+        end
+
+        def to_s
+          @exception.backtrace.find { |line| line !~ IGNORE_REGEX }
+        end
+      end
       # Earlier rubies do not support the two argument form of `caller`. This
       # fallback is logically the same, but slower.
       def self.first_non_rspec_line(*)
-        caller.find { |line| line !~ IGNORE_REGEX }
+        begin
+          raise CallerFinder
+        rescue CallerFinder => error
+          return LazyBacktrace.new(error)
+        end
```

Comment:
I've just benchmarked `raise` against `caller` and `caller_location` and added that to this PR and it seems that although a bare `raise` and capture is faster than both a bare `caller` or a bare `caller_location`, it isn't faster than the skip frames example for `caller_locations`?